clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6176,10 +6176,12 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       // semantic analysis, etc.
       break;
     }
-  } else {
+  } else if (!JA.isDeviceOffloading(Action::OFK_SYCL)) {
     Args.AddLastArg(CmdArgs, options::OPT_fopenmp_simd,
                     options::OPT_fno_openmp_simd);
     Args.AddAllArgs(CmdArgs, options::OPT_fopenmp_version_EQ);
```

Comment:
```suggestion
  } else {
    if (!JA.isDeviceOffloading(Action::OFK_SYCL)
      Args.AddLastArg(CmdArgs, options::OPT_fopenmp_simd,
                      options::OPT_fno_openmp_simd);
    Args.AddAllArgs(CmdArgs, options::OPT_fopenmp_version_EQ);
```
maybe this would be better?

Reply:
@mdtoguchi, I can structure it that way if that is preferable.

Reply:
That would be great, thanks @premanandrao